From 8f1e83bbc31331e3df3195a73ce4aff4797d60fe Mon Sep 17 00:00:00 2001
From: Mike Gerdts <mike.gerdts@joyent.com>
Date: Tue, 8 Jan 2019 16:41:12 +0000
Subject: [PATCH] OS-7488 vm delete fails after OS-7436 Reviewed by: Dave Eddy
 <dave.eddy@joyent.com> Approved by: Dave Eddy <dave.eddy@joyent.com>

---
 src/vm/node_modules/vminfod/vminfod.js | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/src/vm/node_modules/vminfod/vminfod.js b/src/vm/node_modules/vminfod/vminfod.js
index 92a7fed0..b5131232 100644
--- a/src/vm/node_modules/vminfod/vminfod.js
+++ b/src/vm/node_modules/vminfod/vminfod.js
@@ -2314,17 +2314,19 @@ Vminfod.prototype.handleZpoolEvent = function handleZpoolEvent(obj) {
         func: function handleZfsEventTask(extras, cb) {
             var datasetObj = {};
             var uuid;
-            var zpools = self.zonesZpools();
 
             /*
-             * Check to see if the zpool for the event seen is in the list of
-             * zpools in use by VMs on the system.
+             * If this doesn't look like it could be a zonepath dataset, ensure
+             * that the dataset's zpool is one that is used by any VM.
              */
-            if (!zpools.hasOwnProperty(zpool)) {
-                self.log.debug({zpool: zpool, zpools: Object.keys(zpools)},
-                    'ignoring %s zfs event for zpool: %s', dsname, zpool);
-                cb();
-                return;
+            if (parts.length !== 2) {
+                var zpools = self.zonesZpools();
+                if (!zpools.hasOwnProperty(zpool)) {
+                    self.log.debug({zpool: zpool, zpools: Object.keys(zpools)},
+                        'ignoring %s zfs event for zpool: %s', dsname, zpool);
+                    cb();
+                    return;
+                }
             }
 
             if (parts[1]) {
-- 
2.21.0

